pkg/util/threadsafesets/string.go

```diff
@@ -0,0 +1,136 @@
+package threadsafesets
+
+import (
+	"strings"
+	"sync"
+)
+
+// ThreadSafeIgnoreCaseSet is a set of strings that is case-insensitive and thread-safe.
+type ThreadSafeIgnoreCaseSet struct {
+	mu  sync.RWMutex
+	set map[string]struct{}
+}
+
+func (s *ThreadSafeIgnoreCaseSet) Equals(other *ThreadSafeIgnoreCaseSet) bool {
+	if !s.Initialized() || !other.Initialized() {
+		return false
+	}
+
+	if s == other {
+		return true
+	}
+
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	other.mu.RLock()
+	defer other.mu.RUnlock()
```

Comment:
Is this safe? I think it can deadlock if it will be called from both sets at the same time. We should use some property (memory address?) of each set to lock in the same order no matter which set calls equals.

Reply:
What do you mean? Multiple threads can hold RLocks at a time.